### PR TITLE
More User and Review routes

### PR DESCRIPTION
**Pull request by rfmcmillan**
- Users: Added post, delete and put routes
- Reviews: Created get, post, delete and put routes

### CODE REVIEW

File: server/api/reviews.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
+const express = require('express')
+const router = express.Router()
+
+const {
+    models: { User, Review, Product },
+} = require('../db')
+
+router.get('/', async (req, res, next) => {
```
Comment (alejvillanueva) [inline server/api/reviews.js:8]: You might want to include the User so we are able to get the User information

Comment (alejvillanueva) [reply server/api/reviews.js:8]: Same for the route below

File: server/api/users.js [hunk 2]
```diff
@@ -1,23 +1,57 @@
...
+router.get('/', async (req, res, next) => {
+    try {
+        const users = await User.findAll({ include: Review })
+        res.send(users)
+    } catch (error) {
+        next(error)
+    }
+})
+
+router.get('/:id', async (req, res, next) => {
```
Comment (alejvillanueva) [inline server/api/users.js:18]: You should include Review in here as well

File: test/review-model.js [hunk 3]
```diff
@@ -1,132 +1,244 @@
...
+        describe('GET', () => {
+            it('api/reviews', async () => {
+                const response = await app.get('/api/reviews')
+                const reviews = response.body
+                expect(response.status).to.equal(200)
+                expect(reviews).to.be.ok
+                expect(reviews).to.be.an('array')
+            })
+            it('api/reviews/:id', async () => {
+                const jack = await Review.create({
```
Comment (alejvillanueva) [inline test/review-model.js:159]: Should this be User.create()?

File: test/review-model.js [hunk 4]
```diff
@@ -1,132 +1,244 @@
...
+                })
+                const newReview = response.body
+                expect(response.status).to.equal(201)
+                expect(newReview).to.be.ok
+                expect(newReview).to.be.an('object')
+            })
+        })
+        describe('DELETE', () => {
+            it('api/reviews/:id', async () => {
+                const tempReview = await Review.create({
```
Comment (alejvillanueva) [inline test/review-model.js:199]: Should this be user.create or review.create?

**Review by alejvillanueva — APPROVED** [review]
Just a couple of changes needed

**Review by alejvillanueva — CHANGES_REQUESTED** [review]
Just needs a couple adjustments

File: server/api/reviews.js (hunk 1 above)
Comment (rfmcmillan) [reply server/api/reviews.js:8]: Great, I'll include the Product as well just in case

File: server/api/users.js (hunk 2 above)
Comment (rfmcmillan) [reply server/api/users.js:18]: Got it - I'll add that in.

File: test/review-model.js (hunk 4 above)
Comment (rfmcmillan) [reply test/review-model.js:199]: Thank you, fixed that and also changed test to first check that review was initially created and then successfully deleted.

**Comment by rfmcmillan** [comment]
Just pushed my fixes up - sorry for those mistakes in there. I'll start taking more time to double-check everything.

File: server/api/reviews.js (hunk 1 above)
Comment (alejvillanueva) [reply server/api/reviews.js:8]: Perfect, good call!

File: server/api/users.js (hunk 2 above)
Comment (alejvillanueva) [reply server/api/users.js:18]: We can talk about while we are all together but we might also want to include Orders? Maybe something you can @kfless12 can discuss

**Review by alejvillanueva — APPROVED** [review]

**Review by YiruDing — APPROVED** [review]
I'm re-watching the video

**Pull request merged**